sparse_strips/vello_api/src/painter.rs

```diff
@@ -0,0 +1,126 @@
+        &mut self,
+        brush: impl Into<OurBrush>,
+        // We'd like to support both brushes which are "object-local" and brushes which are "scene-local".
+        // Image for example you want a gradient to be applied over a whole paragraph of text.
+        // The naive solution would need to apply the gradient with a paint transform which undoes the
+        // object's transform. That's clearly pretty poor, and we can do better.
+        // However, this isn't exposed in the current Vellos, so we're choosing to defer this.
+        // transform: Affine,
+        paint_transform: Affine,
+    );
+    fn set_blurred_rounded_rect_brush(
+        &mut self,
+        // transform: Affine,
+        paint_transform: Affine,
+        color: Color,
+        rect: &Rect,
+        radius: f32,
+        std_dev: f32,
+    );
+    fn set_solid_brush(&mut self, color: Color) {
+        // The transform doesn't matter for a solid color brush.
+        self.set_brush(Brush::Solid(color), Affine::IDENTITY);
+    }
+
+    fn fill_blurred_rounded_rect(
+        &mut self,
+        transform: Affine,
+        color: Color,
+        rect: &Rect,
+        radius: f32,
+        std_dev: f32,
+    ) {
+        self.set_blurred_rounded_rect_brush(transform, color, rect, radius, std_dev);
+        // The impulse response of a gaussian filter is infinite.
+        // For performance reason we cut off the filter at some extent where the response is close to zero.
+        let kernel_size = (2.5 * std_dev) as f64;
+
+        let shape: Rect = rect.inflate(kernel_size, kernel_size);
+        self.fill_path(transform, Fill::EvenOdd, shape);
+    }
+
+    // This can be accessed through downcasting:
+    // (Well, kind of...; that doesn't apply to Scene, which is by far the most common type here.)
+    // fn set_aliasing_threshold(&mut self, aliasing_threshold: Option<u8>);
+    // TODO: What does this mean?
+    // fn set_blend_mode(&mut self, blend_mode: BlendMode);
+
+    // TODO: Do we want this exposed?
+    // TODO: &impl Shape?
+    // fn push_clip_path(&mut self, path: &kurbo::BezPath);
+    // fn pop_clip_path(&mut self);
+
+    // TODO: We need to support a fill rule here for clip paths...
+    fn push_layer(
+        &mut self,
+        clip_transform: Affine,
+        clip_path: Option<impl Shape>,
+        blend_mode: Option<BlendMode>,
+        opacity: Option<f32>,
+        // mask: Option<Mask>,
+        // filter: Option<Filter>
+    );
+
+    // TODO: Do we need all of these?
+    // TODO: We need to support a fill rule here...
+    fn push_clip_layer(&mut self, clip_transform: Affine, path: impl Shape) {
+        self.push_layer(clip_transform, Some(path), None, None);
+    }
+
+    fn push_blend_layer(&mut self, blend_mode: BlendMode) {
+        self.push_layer(
+            Affine::IDENTITY,
+            // This needing to be turbofished shows a real footgun with the proped "push_layer" API here.
+            None::<BezPath>,
```

Comment:
I don't think we'll solve this in the short-term, but this is definitely worth creating an issue about.